### PR TITLE
[FIXED]  #4664 mqtt retained messages memory leak

### LINKED ISSUE #4664: MQTT - memory leak when server handle a retained message publish in handleRetainedMsg
### What version were you using?

2.10.2

### What environment was the server running in?

All environments affected

### Is this defect reproducible?

STR:
- publish in mqtt protocol some messages with retain flag enabled, incrementing the `sseq` of messages
- keep the same subject for each message
- heaps grows indefinitely as messages are published, and fill the RAM completely

### Given the capability you are leveraging, describe your expectation?

I setup a test to prove the leak in the function `handleRetainedMsg`. I expect that the handler creates a internal subscription only 1 time per subject (if a new messages comes with same subject, then  the same subscription is used). I also expect that when `handleRetainedMsgDel` is called, then subscription related to that subject is deleted
```
func TestMQTTRetainedMsgDel(t *testing.T) {
	o := testMQTTDefaultOptions()
	s := testMQTTRunServer(t, o)
	defer testMQTTShutdownServer(s)
	mc, _ := testMQTTConnect(t, &mqttConnInfo{clientID: "sub", cleanSess: true}, o.MQTT.Host, o.MQTT.Port)
	defer mc.Close()

	c := testMQTTGetClient(t, s, "sub")
	asm := c.mqtt.asm
	var i uint64
        // 3 messages with increasing sseq are published
	for i = 0; i < 3; i++ {
		rf := &mqttRetainedMsgRef{sseq: i}
                //
		asm.handleRetainedMsg("subject", rf)
	}
	asm.handleRetainedMsgDel("subject", 2)
	if asm.sl.count > 0 {
		t.Fatalf("all retained messages subs should be removed, but %d still present", asm.sl.count)
	}
	fmt.Printf("a")
}
```

### Given the expectation, what is the defect you are observing?

in `handleRetainedMsg` invokation, additional subscriptios are created even if the message subject is the same of an already mapped message. The result is that 
- the subscription list grows with a lot of subscriptions containing the same subject
- the heap is filled

example handling 1000 messages with same subject and increasing `sseq`:
image

### PR DESCRIPTION
**Pull request by pricelessrabbit**
fixes #4664

### CODE REVIEW

File: server/mqtt.go [hunk 1]
```diff
@@ -2029,35 +2029,32 @@ func (as *mqttAccountSessionManager) sendJSAPIrequests(s *Server, c *client, acc
...
-			erm.sseq = rm.sseq
-			// Clear the floor
-			erm.floor = 0
-			// If sub is nil, it means that it was removed from sublist following a
-			// network delete. So need to add it now.
-			if erm.sub == nil {
-				erm.sub = &subscription{subject: []byte(key)}
-				as.sl.Insert(erm.sub)
-			}
+	}
```
Comment (levb) [inline server/mqtt.go:2039]: 1/5 keep the `else`, just add the missing `return`; just a very slight optimization, no need to lookup in the new (empty) map?

**Review by levb — CHANGES_REQUESTED** [review]
@pricelessrabbit thank you for tracking this down and the contribution!

one small nit, otherwise LGTM!

File: server/mqtt.go (hunk 1 above)
Comment (pricelessrabbit) [reply server/mqtt.go]: sure, i removed it to make more readable but you're right for the optmization

**Comment by neilalexander** [comment]
Thanks for the contribution but you need to sign off before we can merge. Please take a look at [CONTRIBUTING.md](https://github.com/nats-io/nats-server/blob/main/CONTRIBUTING.md#sign-off) for more info.

**Review by neilalexander — APPROVED** [review]
Thanks so much, LGTM!

**Pull request merged**